docs/code-search/code-navigation/auto_indexing.mdx

```diff
@@ -0,0 +1,320 @@
+# Auto-indexing
+
+<p className="subtitle">Learn and understand how auto-indexing works.</p>
+
+<Callout type="note">This feature is in beta for self-hosted customers.</Callout>
+
+With Sourcegraph deployments supporting [executors](/admin/executors/), your repository contents can be automatically analyzed to produce a code graph index file. Once [auto-indexing is enabled](/code_navigation/how-to/enable_auto_indexing) and [auto-indexing policies are configured](/code_navigation/how-to/configure_auto_indexing), repositories will be periodically cloned into an executor sandbox, analyzed, and the resulting index file will be uploaded back to the Sourcegraph instance.
+
+## Enable Code Navigation auto-indexing
+
+<Callout type="note">This feature is in beta for self-hosted customers.</Callout>
+
+The following docs explains how to turn on [auto-indexing](/code_navigation/explanations/auto_indexing) on your Sourcegraph instance to enable [precise code navigation](/code_navigation/explanations/precise_code_navigation).
+
+### Deploy executors
+
+First, [deploy the executor service](/admin/executors/deploy_executors) targeting your Sourcegraph instance. This will provide the necessary compute resources that clone the target Git repository, securely analyze the code to produce a code graph data index, then upload that index to your Sourcegraph instance for processing.
```

Comment:
@MaedahBatool how do we handle different steps for Cloud vs self-hosted customers in other parts of the docs? Here, the step of 'Deploy executors' only makes sense for self-hosted customers, we handle this ourselves for Cloud customers.

Reply:
@varungandhi-src, there is no set pattern for segregating docs for Cloud vs. Self-hosted audience. The best we can do is add a callout stating this is for self-hosted customers.